### ISSUE TITLE
libuv test fail on `aarch64-linux` and `riscv64-linux`

### ISSUE THREAD

**Issue by GenericNerdyUsername** [NixOS/nixpkgs#190807]
### Steps To Reproduce
1. build `nixpkgs.libuv` when system is `aarch64-linux` or `riscv64-linux` (can be tested on x86 nixos by adding `boot.binfmt.emulatedSystems = [ "aarch64-linux" "riscv64-linux" ]`)

### Build log
aarch64: https://gist.github.com/GenericNerdyUsername/9640b3df1ab98d1f8cfdb75d7fd01a87
riscv64: https://gist.github.com/GenericNerdyUsername/8f8e5f3389001315f6aae4216a9de183

### Notify maintainers
@cstrahan 

### Metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.19.5, NixOS, 22.11 (Raccoon), 22.11.20220901.2da64a8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by danielbarter** [comment]
Don't think this is an architecture issue. I am seeing the exactly the same issue running
```
nix-build -E 'with import <nixpkgs> { config.replaceStdenv = ({ pkgs }: pkgs.clangStdenv);}; libuv'
```

system info:
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.90, NixOS, 22.11 (Raccoon), 22.11.20230204.8e82401`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"nixos-22.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by winterqt** [comment]
This was fixed in #211896, keeping open until -next is merged.

**Comment by danielbarter** [comment]
Ah great. Thanks @winterqt !

**Comment by GenericNerdyUsername** [comment]
> This was fixed in #211896, keeping open until -next is merged.

Did you mean merged into master or merged into a stable branch?

**Comment by Artturin** [comment]
https://nixpk.gs/pr-tracker.html?pr=211896 its in nixos-unstable now
